src/Run.js

```diff
@@ -1,12 +1,12 @@
-function Run(ref, json_q, callback,progressObj) {
-  if(!progressObj) progressObj = {stop: function(){}}
+function Run(ref, json_q, callback, progressObj) {
```

Comment:
Can we open a new PR to rename these parameters something more descriptive like inputImage and things like that - clues to the working of the function?

Reply:
@jywarren Oh sure, opening an issue for this😁

Reply:
@jywarren Actually I embed 3 things inside the input variable, a history array and 2 methods, so returning would actually be difficult here, what say?